### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by shirhatti**

### CODE REVIEW

**Comment by shirhatti** [comment]
Can y'all specifically review the docs TOC? I'll start adding out more content, but at the minimum I want consensus on what we should document to begin with

File: documentation/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+When running a dotnet application differences in diverse local and production environments can make collecting diagnostics artifacts (e.g., logs, traces, process dumps) challenging. dotnet-monitor aims to simplify the process by exposing a consistent REST API regardless of where your application is run.
+
+## Table of contents
+
+- [Setup](./setup.md)
+- [Getting Started](#)
+    - [Running on a local machine](#)
+    - [Running in Docker](#)
+    - [Running in a kubernetes cluster](#)
+- [API Endpoints](#)
```
Comment (jander-msft) [inline documentation/README.md:14]: Need /logs

File: documentation/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# 📖 `dotnet-monitor` documentation
+
+`dotnet-monitor` is a tool that makes it easier to get access to diagnostics information in a dotnet process.
+
+When running a dotnet application differences in diverse local and production environments can make collecting diagnostics artifacts (e.g., logs, traces, process dumps) challenging. dotnet-monitor aims to simplify the process by exposing a consistent REST API regardless of where your application is run.
```
Comment (jander-msft) [inline documentation/README.md:5]: nit: 'HTTP' instead of 'REST'?

File: documentation/setup.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Setup
+
+`dotnet-monitor` is available via two different distribution mechanism:
+
+- As a .NET Core global tool; and
+- As a container image available via the Microsoft Container Registry (MCR)
+
+### .NET Core global tool
```
Comment (jander-msft) [inline documentation/setup.md:8]: This section should be using preview.4 not preview.5

File: documentation/setup.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+### Container image
+
+The latest public preview of the `dotnet-monitor` container image is avaialbe on MCR. You can pull the latest image using the following command:
+
+```cmd
+docker pull mcr.microsoft.com/dotnet/monitor:5.0.0-preview.5
+```
+
+### Working with CI builds
```
Comment (jander-msft) [inline documentation/setup.md:32]: This section should be using preview.5 not preview.6

File: documentation/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    - [Running in Docker](#)
+    - [Running in a kubernetes cluster](#)
+- [API Endpoints](#)
+    - [OpenAPI document](./openapi.json)
+    - [`/processes`](#)
+    - [`/dump`](#)
+    - [`/gcdump`](#)
+    - [`/trace`](#)
+    - [`/metrics`](#)
+- [Configuration](#)
```
Comment (jander-msft) [inline documentation/README.md:21]: I think we'll want to reference the JSON schema under Configuration.

File: documentation/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+- [Authentication](#)
+    - [Windows Auth](#)
+    - [API token Auth](#)
+- [Egress Providers](#)
+- [Clone, build, and test the repo](./building.md)
+- [Official Build Instructions](./official-build-instructions.md)
```
Comment (wiktork) [inline documentation/README.md:27]: Did we want to have a roadmap for some of the planned features?

Comment (shirhatti) [reply documentation/README.md:27]: I'll create a separate issue for that. I don't think it's needed for the preview

Comment (shirhatti) [reply documentation/README.md:27]: https://github.com/dotnet/dotnet-monitor/issues/158

File: documentation/setup.md (hunk 4 above)
Comment (shirhatti) [reply documentation/setup.md:32]: We're releasing preview5, the CI builds would have been bumped up to preview.6 by then?

Comment (jander-msft) [reply documentation/setup.md:32]: We are releasing preview.4 from release/5.0, the main branch is set to preview.5.

Comment (shirhatti) [reply documentation/setup.md:32]: My bad then. I'll fixup the version numbers

File: documentation/setup.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+- As a container image available via the Microsoft Container Registry (MCR)
+
+### .NET Core global tool
+
+The `dotnet-monitor` global tool requires a .NET 3.1 or newer SDK installed as a pre-requisite. If you do not have a new enough SDK, you can install a new one from the [Download .NET webpage](https://dotnet.microsoft.com/download).
+
+The latest public preview of `dotnet-monitor` is available on Nuget. You can download the latest version using the following command:
+
+```cmd
+dotnet tool install -g dotnet-monitor --version 5.0.0-preview.4
```
Comment (jander-msft) [inline documentation/setup.md:15]: 5.0.0-preview.4.*

File: documentation/setup.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+If you already have `dotnet-monitor` installed and want to update:
+
+```cmd
+dotnet tool update -g dotnet-monitor --version 5.0.0-preview.4
```
Comment (jander-msft) [inline documentation/setup.md:21]: 5.0.0-preview.4.*

File: documentation/setup.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+### Container image
+
+The latest public preview of the `dotnet-monitor` container image is avaialbe on MCR. You can pull the latest image using the following command:
+
+```cmd
+docker pull mcr.microsoft.com/dotnet/monitor:5.0.0-preview.4
```
Comment (jander-msft) [inline documentation/setup.md:29]: FYI, this version is correct, it does not have anything beyond the `preview.4` portion.

File: documentation/setup.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+### Working with CI builds
+
+In addition to public previews, we also publish last-known-good (LKG) builds for the next release of `dotnet-monitor`.
+
+The LKG build of `dotnet-monitor` is available on a private package feed. You can download the latest versionof the .NET global tool using the following command:
+
+```cmd
+dotnet tool install -g dotnet-monitor --add-source https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet-tools/nuget/v3/index.json --version 5.0.0-preview.5.*
```
Comment (jander-msft) [inline documentation/setup.md:39]: 5.0.0-preview.5.*

File: documentation/setup.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+If you already have `dotnet-monitor` installed and want to update:
+
+```cmd
+dotnet tool update -g dotnet-monitor --add-source https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet-tools/nuget/v3/index.json --version 5.0.0-preview.5.*
```
Comment (jander-msft) [inline documentation/setup.md:45]: 5.0.0-preview.5.*

File: documentation/setup.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+The LKG build of `dotnet-monitor` is also available as a container image. You can pull the image using the following command:
+
+```cmd
+docker pull mcr.microsoft.com/dotnet/nightly/monitor:5.0.0-preview.5
```
Comment (jander-msft) [inline documentation/setup.md:51]: FYI, this version is correct, it does not have anything beyond the `preview.5` portion.

File: documentation/README.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+## Table of contents
+
+- [Setup](./setup.md)
+- [Getting Started](#)
+    - [Running on a local machine](#)
+    - [Running in Docker](#)
+    - [Running in a kubernetes cluster](#)
+- [API Endpoints](#)
+    - [OpenAPI document](./openapi.json)
+    - [`/processes`](#)
```
Comment (jander-msft) [inline documentation/README.md:16]: Should the routes be alphabetized?

File: documentation/README.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+    - [`/dump`](#)
+    - [`/gcdump`](#)
+    - [`/trace`](#)
+    - [`/metrics`](#)
+    - [`/logs`](#)
+- [Configuration](#)
+    - [JSON Schema](./schema.json)
+- [Authentication](#)
```
Comment (jander-msft) [inline documentation/README.md:24]: I think these should be "Windows" and "API Key". No need to repeat Auth each time and the API Key authentication is key based, not token based.

**Comment by jander-msft** [comment]
I'll take a crack at documentation one of the routes (e.g. `/processes`), get that reviewed, and then we can use that style as a template for the rest of routes.

File: documentation/README.md (hunk 13 above)
Comment (shirhatti) [reply documentation/README.md:16]: I'm not a fan of alphabetizing the routes. I ordered them in a way that made sense to me

**Review by willpe — APPROVED** [review]

**Pull request merged**